### DISCUSSION TITLE
How to disable <Edit Chart> in chart controls?

### DISCUSSION THREAD

**Discussion by aSingleDragon** [apache/superset#25642] [Q&A / Help]
I have embedded a superset dashboard in my system, but I don't want users to see the "edit chart", "view SQL" buttons. What should I do? What role permissions need to be modified?
<img width="519" alt="image" src="https://github.com/apache/superset/assets/22795018/abdd7a40-376b-4107-b8b0-5567cd084dec">

**Comment by MicroUiChandan** [comment]
You can use this css for hide Edit Chart Option

ul.ant-dropdown-menu-vertical li:nth-child(4) {
  display: none;
}

**Reply by rajubish** [reply]
Is there any role permission to disable/remove view query and edit chart options?

**Reply by MicroUiChandan** [reply]
By removing the "Can Explore on superset" privilege, we can get rid of the View Query option. Just a heads-up, the side effect is that it will also hide the View as Table feature.

**Reply by jjcorredor99** [reply]
@MicroUiChandan Thank you! This information has been very helpful. Do you happen to know where we can find detailed information about each permission and its effects

**Reply by inchara27** [reply]
I have embedded a superset dashboard in my system, but I don't want users to see the "Share", "view SQL" buttons. What should I do? What role permissions need to be modified?
![02032024](https://github.com/apache/superset/assets/48905512/b25eec11-e014-47fd-9e2d-80a04a89193d)

**Reply by mounikagancheeri** [reply]
Tried this but only able to hide edit chart option and for admin as well this change is getting effected if I am removing can explore on superset for gamma level

**Reply by XwyFromCN** [reply]
it works，thanks！

**Reply by Pavel-Malai** [reply]
Beware that this css selector will also affect the css in the superset admin app, for example it will hide menu items from the edit dashboard menu 
![image](https://github.com/user-attachments/assets/7f4a9ecd-7278-4671-aeae-9ed6a2823f13)

**Comment by inchara27** [comment]
> I have embedded a superset dashboard in my system, but I don't want users to see the "edit chart", "view SQL" buttons. What should I do? What role permissions need to be modified? <img alt="image" width="519" src="https://private-user-images.githubusercontent.com/22795018/275183847-abdd7a40-376b-4107-b8b0-5567cd084dec.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkzNzI0NDYsIm5iZiI6MTcwOTM3MjE0NiwicGF0aCI6Ii8yMjc5NTAxOC8yNzUxODM4NDctYWJkZDdhNDAtMzc2Yi00MTA3LWI4YjAtNTU2N2NkMDg0ZGVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzAyVDA5MzU0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM4ZTQ5NzAwMzE0YmYwZjgzZTZkOGU4ZWIyNzg0ZGM3NDhkMmI4MDVkYzYwN2U0NDI0YTlkNzZhMTJlYWExNDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NTVui6UM1n1Fyd3adndXrvEqDQVEw9yW-866Bm6NLT0">

But how did you remove Share options

**Reply by VikramKumarArammeem** [reply]
We can create custom role with these set of permission. I got this from Superset community slack channel shared by Sam Firke - `can read on CssTemplate, can read on Chart, can read on Annotation, can read on Dashboard, can get on OpenApi, can write on DashboardFilterStateRestApi, can read on DashboardFilterStateRestApi, can write on DashboardPermalinkRestApi, can read on DashboardPermalinkRestApi, can write on ExploreFormDataRestApi, can read on ExploreFormDataRestApi, can write on ExplorePermalinkRestApi, can read on ExplorePermalinkRestApi, can list on FilterSets, can time range on Api, can query on Api, can query form data on Api, can filter on Superset, can slice json on Superset, can explore json on Superset, can validate sql json on Superset, can favstar on Superset, can dashboard permalink on Superset, can sql json on Superset, can queries on Superset, can csv on Superset, can dashboard on Superset, can slice on Superset, can annotation json on Superset, can share dashboard on Superset, can read on AdvancedDataType`

**Reply by marianysilva** [reply]
@VikramKumarArammeem Can you explain how it could help? I didn't get the expected result.

**Comment by SuperMAB** [comment]
I posted a solution here
https://github.com/apache/superset/discussions/25436#discussioncomment-13723809
The embeding SDK seems to grant "Gamma" Role automatically even if you force it to another role
Good Luck

**Reply by marianysilva** [reply]
Hey @SuperMAB you need to setup the [GUEST_ROLE_NAME](https://github.com/apache/superset/blob/master/superset/config.py#L2054) in your config.py file, so you can use a custom role.
